core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -185,6 +198,16 @@ List<ManifestFile> filterManifests(Schema tableSchema, List<ManifestFile> manife
       return ImmutableList.of();
     }
 
+    // Use the current set of referenced manifests as a source of truth when it's a subset of all
+    // manifests and all removals which were performed reference manifests.
+    // If a manifest is not in the trusted referenced set, filtering can be skipped
```

Comment:
I'd clarify what "filtering" means on this line. Are we talking about opening up a manifest to see if it contains deletes? Or something else?

Reply:
Added some clarification on the implication of a manifest not being in the trusted set